### DISCUSSION TITLE
Keeping or removing pyrolyzed vegetation

### DISCUSSION THREAD

**Discussion by ahmedahmed2056** [firemodels/fds#12777] [General]
![image](https://github.com/firemodels/fds/assets/165714562/1f5b5996-0c2d-4cb0-ad89-98ae378af132)

This is the paper of Mell et al. 2009 of burning Douglas fir tree , in their paper they mentioned ( In order to assess the importance of this
effect on model predictions the removal of a pyrolyzed fuel element was handled in two ways. In the first approach, once pyrolysis
is complete, leaving only char, a fuel element remains for the duration of the simulation as a source of drag, thermal mass, and radiative absorbion/emission. In the second approach, once the virgin fuel is fully pyrolyzed it is removed from the computational domain.)

So how to apply those two approaches in this input file 
&HEAD CHID='tree_2_m_14_pc', TITLE='2 m Douglas Fir with 14% moisture content' /

&MESH IJK=24,24,24, XB=-1.2,1.2,-1.2,1.2,0.0,2.4, MULT_ID='mesh' /
&MULT ID='mesh', DX=2.4, DY=2.4, DZ=2.4, I_UPPER=1, J_UPPER=0, K_UPPER=3 /

&TIME T_END=30. / 

&VENT MB='XMIN', SURF_ID='OPEN' /
&VENT MB='XMAX', SURF_ID='OPEN' /
&VENT MB='YMIN', SURF_ID='OPEN' /
&VENT MB='YMAX', SURF_ID='OPEN' /
&VENT MB='ZMAX', SURF_ID='OPEN' /
&VENT PBZ=0.,    SURF_ID='FOIL' /

&SURF ID='FOIL'
      COLOR = 'SILVER'
      MATL_ID='ALUMINUM','CONCRETE'
      THICKNESS(1:2) = 0.0002,0.01 /

&MATL ID='ALUMINUM'
      DENSITY = 2707.
      SPECIFIC_HEAT = 0.896
      CONDUCTIVITY = 204.
      EMISSIVITY = 0.1 /

&MATL ID='CONCRETE'
      DENSITY = 1500.
      SPECIFIC_HEAT = 1.
      CONDUCTIVITY = 0.3  /

&SURF ID                        = 'foliage'
      MATL_ID(1,1) = 'GENERIC VEGETATION'
      MATL_MASS_FRACTION(1,1) = 1.0
      MOISTURE_FRACTION         = 0.14
      SURFACE_VOLUME_RATIO      = 3940.
      LENGTH                    = 0.05
      GEOMETRY                  = 'CYLINDRICAL' /

&SURF ID                        = 'small roundwood'
      MATL_ID(1,1) = 'GENERIC VEGETATION'
      MATL_MASS_FRACTION(1,1) = 1.0
      MOISTURE_FRACTION         = 0.14
      SURFACE_VOLUME_RATIO      = 2667.
      LENGTH                    = 0.10
      GEOMETRY                  = 'CYLINDRICAL' /

&SURF ID                        = 'medium roundwood'
      MATL_ID(1,1) = 'GENERIC VEGETATION'
      MATL_MASS_FRACTION(1,1) = 1.0
      MOISTURE_FRACTION         = 0.14
      SURFACE_VOLUME_RATIO      = 888.
      LENGTH                    = 0.10
      GEOMETRY                  = 'CYLINDRICAL' /

&SURF ID                        = 'large roundwood'
      MATL_ID(1,1) = 'GENERIC VEGETATION'
      MATL_MASS_FRACTION(1,1) = 1.0
      MOISTURE_FRACTION         = 0.14
      SURFACE_VOLUME_RATIO      = 500.
      LENGTH                    = 0.10
      GEOMETRY                  = 'CYLINDRICAL' /

&REAC FUEL='FUEL VAPOR', C=2.10, H=6.20, O=2.16, SOOT_YIELD=0.01, HEAT_OF_COMBUSTION=17425., IDEAL=T /

&SPEC ID='WATER VAPOR' /

&MATL ID                    = 'GENERIC VEGETATION'
      DENSITY               = 500.
      CONDUCTIVITY          = 0.2
      SPECIFIC_HEAT_RAMP    = 'c_v'
      A                     = 1040.
      E                     = 61041.
      NU_SPEC               = 0.75
      SPEC_ID               = 'FUEL VAPOR'
      NU_MATL               = 0.25
      MATL_ID               = 'CHAR'
      HEAT_OF_REACTION      = 418. /

&MATL ID                    = 'CHAR'
      DENSITY               = 300.
      CONDUCTIVITY          = 0.052
      SPECIFIC_HEAT_RAMP    = 'c_v'
      SURFACE_OXIDATION_MODEL = T
      A                  = 465.
      E                  = 68000. 
      SPEC_ID               = 'PRODUCTS','AIR'
      NU_SPEC               = 8.13,-7.17
      MATL_ID               = 'ASH'
      NU_MATL               = 0.04
      HEAT_OF_REACTION      = -25000. /

&MATL ID                    = 'ASH'
      DENSITY               = 67.
      CONDUCTIVITY          = 0.1
      SPECIFIC_HEAT_RAMP    = 'c_v' /

&RAMP ID='c_v', T=  0., F=1.1 /
&RAMP ID='c_v', T=200., F=2.0 /
&RAMP ID='c_v', T=800., F=2.0 /
&PART ID='foliage', DRAG_COEFFICIENT=2.8, SAMPLING_FACTOR=10, SURF_ID='foliage'
      QUANTITIES='PARTICLE TEMPERATURE','PARTICLE MASS', STATIC=.TRUE., COLOR='FOREST GREEN' /
&INIT PART_ID='foliage', XYZ=0.0,0.0,0.15, RADIUS=0.85, HEIGHT=1.9, SHAPE='CONE', N_PARTICLES_PER_CELL=1, DRY=T, MASS_PER_VOLUME=1.64 /

&PART ID='small roundwood', DRAG_COEFFICIENT=2.8, SAMPLING_FACTOR=10, SURF_ID='small roundwood'
      QUANTITIES='PARTICLE TEMPERATURE','PARTICLE MASS','PARTICLE DIAMETER', STATIC=.TRUE., COLOR='BROWN' /
&INIT PART_ID='small roundwood', XYZ=0.0,0.0,0.15, RADIUS=0.85, HEIGHT=1.9, SHAPE='CONE', N_PARTICLES_PER_CELL=1, DRY=T, MASS_PER_VOLUME=0.2827 /

&PART ID='medium roundwood', DRAG_COEFFICIENT=2.8, SAMPLING_FACTOR=10, SURF_ID='medium roundwood'
      QUANTITIES='PARTICLE TEMPERATURE','PARTICLE MASS','PARTICLE DIAMETER', STATIC=.TRUE., COLOR='BROWN' /
&INIT PART_ID='medium roundwood', XYZ=0.0,0.0,0.15, RADIUS=0.85, HEIGHT=1.9, SHAPE='CONE', N_PARTICLES_PER_CELL=1, DRY=T, MASS_PER_VOLUME=0.257 /

&PART ID='large roundwood', DRAG_COEFFICIENT=2.8, SAMPLING_FACTOR=10, SURF_ID='large roundwood'
      QUANTITIES='PARTICLE TEMPERATURE','PARTICLE MASS','PARTICLE DIAMETER', STATIC=.TRUE., COLOR='BROWN' /
&INIT PART_ID='large roundwood', XYZ=0.0,0.0,0.15, RADIUS=0.85, HEIGHT=1.9, SHAPE='CONE', N_PARTICLES_PER_CELL=1, DRY=T, MASS_PER_VOLUME=0.3855 /

&PART ID='ignitor', SAMPLING_FACTOR=1, SURF_ID='burner', STATIC=.TRUE. /
&SURF ID='burner', TMP_FRONT=1500., EMISSIVITY=1., HRRPUA=60., RAMP_T='ignite', RAMP_Q='ignite', RADIUS=0.01, GEOMETRY='SPHERICAL' /
&RAMP ID='ignite', T= 0.0, F=0. /
&RAMP ID='ignite', T= 0.1, F=1. /
&RAMP ID='ignite', T=10.0, F=1. /
&RAMP ID='ignite', T=10.1, F=0. /
&INIT PART_ID='ignitor', XYZ=0.0,0.0,0.16, RADIUS=0.40, SHAPE='RING', N_PARTICLES=400  /

&DUMP DT_HRR=1., DT_DEVC=1. /

&DEVC XB=-1.0,1.0,-1.0,1.0,0.0,4.0, QUANTITY='MPUV', PART_ID='foliage', ID='foliage', SPATIAL_STATISTIC='VOLUME INTEGRAL'  /
&DEVC XB=-1.0,1.0,-1.0,1.0,0.0,4.0, QUANTITY='MPUV', PART_ID='small roundwood', ID='small roundwood', SPATIAL_STATISTIC='VOLUME INTEGRAL'  /
&DEVC XB=-1.0,1.0,-1.0,1.0,0.0,4.0, QUANTITY='MPUV', PART_ID='medium roundwood', ID='medium roundwood', SPATIAL_STATISTIC='VOLUME INTEGRAL'  /
&DEVC XB=-1.0,1.0,-1.0,1.0,0.0,4.0, QUANTITY='MPUV', PART_ID='large roundwood', ID='large roundwood', SPATIAL_STATISTIC='VOLUME INTEGRAL'  /

&DEVC XB=-1.0,1.0,-1.0,1.0,0.0,4.0, OUTPUT=F, QUANTITY='MASS FLUX', PART_ID='foliage', ID='foliage mlr', SPATIAL_STATISTIC='SURFACE INTEGRAL'/
&DEVC XB=-1.0,1.0,-1.0,1.0,0.0,4.0, OUTPUT=F, QUANTITY='MASS FLUX', PART_ID='small roundwood', ID='small roundwood mlr', SPATIAL_STATISTIC='SURFACE INTEGRAL'/
&DEVC XB=-1.0,1.0,-1.0,1.0,0.0,4.0, OUTPUT=F, QUANTITY='MASS FLUX', PART_ID='medium roundwood', ID='medium roundwood mlr', SPATIAL_STATISTIC='SURFACE INTEGRAL'/
&DEVC XB=-1.0,1.0,-1.0,1.0,0.0,4.0, OUTPUT=F, QUANTITY='MASS FLUX', PART_ID='large roundwood', ID='large roundwood mlr', SPATIAL_STATISTIC='SURFACE INTEGRAL'/

&CTRL ID='mlr sum',FUNCTION_TYPE='SUM',INPUT_ID='foliage mlr','small roundwood mlr','medium roundwood mlr','large roundwood mlr'/
&DEVC ID='total mlr',CTRL_ID='mlr sum' QUANTITY='CONTROL VALUE'/

&DEVC XYZ=0.50,0.00,0.2, ORIENTATION= 0.000, 0.000,1, QUANTITY='GAUGE HEAT FLUX GAS', ID='HF1'  /
&DEVC XYZ=1.50,0.00,0.2, ORIENTATION= 0.000, 0.000,1, QUANTITY='GAUGE HEAT FLUX GAS', ID='HF2'  /
&DEVC XYZ=2.50,0.00,0.2, ORIENTATION= 0.000, 0.000,1, QUANTITY='GAUGE HEAT FLUX GAS', ID='HF3'  /
&DEVC XYZ=1.93,0.52,0.2, ORIENTATION=-0.966,-0.259,0, QUANTITY='GAUGE HEAT FLUX GAS', ID='HF4'  /
&DEVC XYZ=1.93,0.52,1.2, ORIENTATION=-0.966,-0.259,0, QUANTITY='GAUGE HEAT FLUX GAS', ID='HF6'  /
&DEVC XYZ=1.93,0.52,2.3, ORIENTATION=-0.966,-0.259,0, QUANTITY='GAUGE HEAT FLUX GAS', ID='HF8'  /
&DEVC XYZ=1.93,0.52,3.4, ORIENTATION=-0.966,-0.259,0, QUANTITY='GAUGE HEAT FLUX GAS', ID='HF9'  /
&DEVC XYZ=1.93,0.52,4.5, ORIENTATION=-0.966,-0.259,0, QUANTITY='GAUGE HEAT FLUX GAS', ID='HF10' /
&DEVC XYZ=2.90,0.78,0.2, ORIENTATION=-0.966,-0.259,0, QUANTITY='GAUGE HEAT FLUX GAS', ID='HF12' /
&DEVC XYZ=2.90,0.78,1.2, ORIENTATION=-0.966,-0.259,0, QUANTITY='GAUGE HEAT FLUX GAS', ID='HF13' /
&DEVC XYZ=2.90,0.78,2.3, ORIENTATION=-0.966,-0.259,0, QUANTITY='GAUGE HEAT FLUX GAS', ID='HF14' /
&DEVC XYZ=2.90,0.78,3.4, ORIENTATION=-0.966,-0.259,0, QUANTITY='GAUGE HEAT FLUX GAS', ID='HF15' /
&DEVC XYZ=2.90,0.78,4.5, ORIENTATION=-0.966,-0.259,0, QUANTITY='GAUGE HEAT FLUX GAS', ID='HF16' /

&SLCF PBY=0., QUANTITY='RADIATION LOSS', CELL_CENTERED=.TRUE. /
&SLCF PBY=0., QUANTITY='INTEGRATED INTENSITY', CELL_CENTERED=.TRUE. /
&SLCF PBY=0., QUANTITY='TEMPERATURE', CELL_CENTERED=.TRUE. /
&SLCF PBY=0., QUANTITY='TEMPERATURE', VECTOR=.TRUE. /

&TAIL /

**Comment by ericvmueller** [comment]
A way to mimic this removal of char with the current code would be to set `EMBER_PARTICLE=T` on the `PART` line. You can then set `EMBER_DENSITY_THRESHOLD` to the value of density for which your particle should be removed and `EMBER_VELOCITY_THRESHOLD` to zero if you want particles to be removed regardless of surrounding velocity. Lastly, if you set `TRACK_EMBERS=F` the particles will not move around the domain as embers but simply be removed from the calculation, like in the paper. 

Check out section 17.2.4 in the User's Guide.

**Reply by ahmedahmed2056** [reply]
Hello Dr Eric,
I tried to put the EMBER_DENSITY_THRESHOLD = 300. which is the char density and EMBER_VELOCITY_THRESHOLD = 0.0  and TRACK_EMBERS=F  and kept the drag coefficient 2.8. The resulting curve is not correct. So any other suggestions? your help will be highly appreciated.

Quoting again from Mell 2009 ( As stated above, post-burn observations of the experiments
determined that the foliage and smaller roundwood (<10 mm)
for the drier 2 m tall trees and the 5 m tall trees were removed
from the tree during burning. This process was not directly modeled but can affect heat transfer within the crown by altering the
fluid flow (via drag) and radiant heat fluxes (via changes in radiation
absorption/emission). In order to assess the importance of this
affect on model predictions the removal of a pyrolyzed fuel element
was handled in two ways. In the first approach, once pyrolysis
is complete, leaving only char, a fuel element remains for the
duration of the simulation as a source of drag, thermal mass, and
radiative absorbion/emission. In this case, the equation for the fuel
element temperature (Eq. (25) in the Appendix) is solved without
the thermal degradation terms. Both r and b retain their original
values during the post-pyrolysis phase since the charred vegetation
is assumed to have the shape and size of the virgin fuel. In
the second approach, once the virgin fuel is fully pyrolyzed it is removed
from the computational domain.]

![image](https://github.com/firemodels/fds/assets/165714562/e2d7e807-daba-4a3f-b8aa-c3484bab3b8a)

**Reply by ericvmueller** [reply]
It looks like the peak is not that far off but the total mass consumed is maybe too low in your results. Keep in mind the uncertainty bars in the mass loss rate and the variability of the total mass consumed (Figure 7). 
There are a number of things which are not identical in the input you pasted above and the 2009 paper. For example, the text you quote says the size of the particles does not change as they burn, but I don't think that's the case in your input. The drag model is different. The pyrolysis models are different. Etc. A lot of these topics are areas of active research, but you could investigate the sensitivity.

If you look at the thread here: https://github.com/firemodels/fds/issues/11507, you will also see things can be very sensitive to how the ignition is modeled.

**Reply by ahmedahmed2056** [reply]
Dear Dr Meueller,
Thanks for your reply. 
1) When you said that the drag and pyrolysis models are different, did you mean emerging new models of the subsequent FDS versions until today? or is the problem in my input for these models?  

2) I used the [NIST_Douglas_Firs](https://github.com/firemodels/fds/tree/master/Validation/NIST_Douglas_Firs) input files from your uploads on GitHub as I didn't find executable details of the pyrolysis and the drag in Mell's paper. I found the reference temperature for pyrolysis and when I used it didn't give a good result, especially with the extensively defining the moisture (instead of using the moisture fraction in MATL directly). for the drag coefficient, the paper mentioned that it is a function of the Reynolds which I don't know how to input in the FDS input model. Do you encourage me to officially ask for Mell's input file to learn from it?
